kiwixbuild/dependencies/libzim.py

```diff
@@ -29,3 +31,15 @@ def configure_option(self):
             if platformInfo.name == "flatpak":
                 return "--wrap-mode=nodownload"
             return ""
+
+        def _install(self, context):
+            super()._install(context)
+            platformInfo = self.buildEnv.platformInfo
+            # strip build-path info from installed libzim.dylib
+            if platformInfo.name == 'native_mixed' and option('target') == 'libzim' and neutralEnv('distname') == "Darwin":
+                for lib in pathlib.Path(self.buildEnv.install_dir).glob("**/libzim.*.dylib"):
+                    if lib.is_symlink():
+                        continue
+                    command = "install_name_tool -id {name} {path}".format(name=lib.name, path=str(lib))
```

Comment:
I'm not sure about this.
It should be in the libzim repository.
kiwix-build should ideally only install dependencies and launch the build system "as it should be launch". If kiwix-build has to launch `install_name_tool` it means that it is part of the normal way to compile libzim (and so, at least, being documented in libzim as the normal way to build libzim on macos).

On top of that, it seems that meson fix the rpath (and so call `install_name_tool`on mac) at installation. If this is not the case, this is maybe an bug (in libzim or meson).

Reply:
Well that's the thing, meson does fix it at install-time, by setting it to the install's path.

```
reg@homelet:[~/src/kiwix-build]:(macos-dist)  $ otool -L BUILD_native_mixed/INSTALL/lib/libzim.6.dylib
BUILD_native_mixed/INSTALL/lib/libzim.6.dylib:
	/Users/reg/src/kiwix-build/BUILD_native_mixed/INSTALL/lib/libzim.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
```

You see that the install path (`/Users/reg/src/kiwix-build/BUILD_native_mixed/INSTALL/lib/`) is present in the rpath.

meson is right in doing so as we are requesting it to install the lib.

But we want to distribute that file to users who will not install it or install it elsewhere. So this removal is a special step for this particular distribution of a *static/standalone* binary.

I am not sure about the implications of doing it in-meson.

Reply:
I don't know what must be done on macos.

But what you describe is a special case of the distribution of a dynamic library.
It should not be handle here, else we will remove the rpath everytime a user use kiwix-build to build libzim on macos. Either it what we want and we must move this on meson side. Or we don't and this must be moved on the distribution steps (when we create the binary archive and publish it, somewhere in `.github/scripts/common.py` or `.github/scripts/build_release_nightly.py`)